helix-core/src/main/java/org/apache/helix/model/CloudConfig.java

```diff
@@ -52,30 +55,23 @@
 
   /**
    * Instantiate the CloudConfig for the cloud
-   * @param cluster
-   */
-  public CloudConfig(String cluster) {
-    super(cluster);
-  }
-
-  /**
-   * Instantiate with a pre-populated record
-   * @param record a ZNRecord corresponding to a cloud configuration
    */
-  public CloudConfig(ZNRecord record) {
-    super(record);
+  private CloudConfig() {
+    super(CLOUD_CONFIG_KW);
   }
 
   /**
    * Instantiate the config using each field individually.
    * Users should use CloudConfig.Builder to create CloudConfig.
-   * @param cluster
    * @param enabled
+   * @param cloudProvider
    * @param cloudID
+   * @param cloudInfoSource
+   * @param cloudProcessorName
    */
-  public CloudConfig(String cluster, boolean enabled, CloudProvider cloudProvider, String cloudID,
+  private CloudConfig(boolean enabled, CloudProvider cloudProvider, String cloudID,
```

Comment:
Let's make it more generic. In the future, if we have more entries, we may need change the constructor for adding new arguments.

My suggestion will be taking the ZnRecord as the input instead of having all the entries as arguments. Also that helps to reduce the place containing specific entries copying logic.

Reply:
Fixed. Thanks for the suggestion.